lib/puppet_blacksmith/forge.rb

```diff
@@ -65,27 +62,71 @@ def push!(name, package = nil)
 
     private
 
-    def load_credentials_from_file
-      credentials_file = File.expand_path("~/.puppetforge.yml")
-      unless File.exists?(credentials_file)
+    def load_credentials
+      file_credentials = load_credentials_from_file
+      env_credentials = load_credentials_from_env
+
+      credentials = DEFAULT_CREDENTIALS.merge file_credentials
+      credentials = credentials.merge env_credentials
+
+      self.username = credentials['username'] if credentials['username']
+      self.password = credentials['password'] if credentials['password']
+      if credentials['forge']
+        # deprecated
+        puts "'forge' entry is deprecated in .puppetforge.yml, use 'url'"
+        self.url = credentials['forge']
+      end
+      self.url = credentials['url'] if credentials['url']
+
+      unless self.username && self.password
         raise Blacksmith::Error, <<-eos
-Could not find Puppet Forge credentials file '#{credentials_file}'
-Please create it
+Could not find Puppet Forge credentials!
+
+Please set the environment variables
+BLACKSMITH_FORGE_URL
+BLACKSMITH_FORGE_USERNAME
+BLACKSMITH_FORGE_PASSWORD
+
+or create the file '#{CREDENTIALS_FILE}'
+with content similiar to:
+
 ---
 url: https://forgeapi.puppetlabs.com
 username: myuser
 password: mypassword
+
     eos
       end
-      credentials = YAML.load_file(credentials_file)
-      self.username = credentials['username']
-      self.password = credentials['password']
-      if credentials['forge']
-        # deprecated
-        puts "'forge' entry is deprecated in .puppetforge.yml, use 'url'"
-        self.url = credentials['forge']
+    end
+
+    def load_credentials_from_file
+      file = File.expand_path(CREDENTIALS_FILE)
+
+      if File.exists?(file)
+        credentials = YAML.load_file(CREDENTIALS_FILE)
```

Comment:
shouldn't this be `YAML.load_file(file)`? 

I just tried your path with environment variables set and the credentials file still there and got this: 
```
Errno::ENOENT: No such file or directory @ rb_sysopen - ~/.puppetforge.yml
/var/opt/jenkins/.gem/ruby/2.1.5/gems/puppet-3.8.5/lib/puppet/vendor/safe_yaml/lib/safe_yaml.rb:188:in `initialize'
/var/opt/jenkins/.gem/ruby/2.1.5/gems/puppet-3.8.5/lib/puppet/vendor/safe_yaml/lib/safe_yaml.rb:188:in `open'
/var/opt/jenkins/.gem/ruby/2.1.5/gems/puppet-3.8.5/lib/puppet/vendor/safe_yaml/lib/safe_yaml.rb:188:in `unsafe_load_file'
/var/opt/jenkins/.gem/ruby/2.1.5/gems/puppet-3.8.5/lib/puppet/vendor/safe_yaml/lib/safe_yaml.rb:153:in `load_file_with_options'
/var/opt/jenkins/.gem/ruby/2.1.5/bundler/gems/puppet-blacksmith-92151cf9cc73/lib/puppet_blacksmith/forge.rb:106:in `load_credentials_from_file'
/var/opt/jenkins/.gem/ruby/2.1.5/bundler/gems/puppet-blacksmith-92151cf9cc73/lib/puppet_blacksmith/forge.rb:66:in `load_credentials'
/var/opt/jenkins/.gem/ruby/2.1.5/bundler/gems/puppet-blacksmith-92151cf9cc73/lib/puppet_blacksmith/forge.rb:19:in `initialize'
/var/opt/jenkins/.gem/ruby/2.1.5/bundler/gems/puppet-blacksmith-92151cf9cc73/lib/puppet_blacksmith/rake_tasks.rb:75:in `new'
/var/opt/jenkins/.gem/ruby/2.1.5/bundler/gems/puppet-blacksmith-92151cf9cc73/lib/puppet_blacksmith/rake_tasks.rb:75:in `block (2 levels) in define'
```
When I remove the credentials file it works